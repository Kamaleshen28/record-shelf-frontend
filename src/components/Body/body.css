

.body-content{
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-content: center;
    gap:10px;
    align-items: center;
}
.sync-button{
    background-color: #4C63AA;
    color: white;
    font-weight: bolder;
    width:120px;
    height:40px;
    border-radius: 20px;
    border: none;
    font-weight: bolder;
    font-size: 20px;
    cursor: pointer;
}

.body-content-container{
    font-weight: lighter;
    font-size: 25px;
}