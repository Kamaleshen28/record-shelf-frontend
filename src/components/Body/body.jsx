import React from 'react';
import { useNavigate } from 'react-router-dom';
import './body.css';

export default function Body(){
    const navigate = useNavigate();

    const syncButtonClickHandler = () => {
        navigate('/allSongs');
    };

    return(
        <div className='body-content'>
            <div className='body-content-container'>
                <p>:((</p>
                <p>seems a bit empty in here..</p>
            </div>
            <div className='button-container'>
                <button onClick={syncButtonClickHandler} className='sync-button' data-testid='sync-button' >sync</button>
            </div>
        </div>
    );
}