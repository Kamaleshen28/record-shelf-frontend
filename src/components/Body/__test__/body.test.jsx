import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import Body from '../body';
import { useNavigate } from 'react-router-dom';

jest.mock('react-router-dom');


describe('Body', () => {
    it('should render body correctly',async () => {
        const {asFragment} = render(<Body />);
        expect(asFragment()).toMatchSnapshot();

    });

    it('should navigate to /allSongs when the sync button is clicked', async () => {
        const navigateFn = jest.fn();
        useNavigate.mockReturnValue(navigateFn);
        render(<Body />);
        const syncButton = screen.getByTestId('sync-button');
        fireEvent.click(syncButton);
        expect(navigateFn).toBeCalledWith('/allSongs');
    });
});


