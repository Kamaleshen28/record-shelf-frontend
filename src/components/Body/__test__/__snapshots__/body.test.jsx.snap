// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body should render body correctly 1`] = `
<DocumentFragment>
  <div
    class="body-content"
  >
    <div
      class="body-content-container"
    >
      <p>
        :((
      </p>
      <p>
        seems a bit empty in here..
      </p>
    </div>
    <div
      class="button-container"
    >
      <button
        class="sync-button"
        data-testid="sync-button"
      >
        sync
      </button>
    </div>
  </div>
</DocumentFragment>
`;
